### PR TITLE
Move cost models from incident to incident types

### PR DESCRIPTION
**Pull request by metroid-samus**
- Adds `cost_model` field to `IncidentType`.
- Removes `cost_model` field from `Incident`.
- Updates cost model and incident type documentation.
- Immediately recalculates incident costs if either of the criteria is met:
  - An incident changes its `incident_type`
  - An `IncidentType` changes its `cost_model`.
  
  This change requires a database upgrade.

### CODE REVIEW

File: src/dispatch/database/revisions/tenant/versions/2024-04-22_50e99c66e72f.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+"""empty message
```
Comment (kevgliss) [inline src/dispatch/database/revisions/tenant/versions/2024-04-22_50e99c66e72f.py:1]: nit: Consider adding a description

File: src/dispatch/database/revisions/core/versions/2024-04-22_83ec59cd3e4c.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+
+# revision identifiers, used by Alembic.
+revision = '83ec59cd3e4c'
+down_revision = 'ed0b0388fa3f'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
```
Comment (kevgliss) [inline src/dispatch/database/revisions/core/versions/2024-04-22_83ec59cd3e4c.py:18]: I'm not really sure any of these are actually needed? As they look like they are just removing and re-added.

File: src/dispatch/database/revisions/tenant/versions/2024-04-22_50e99c66e72f.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# revision identifiers, used by Alembic.
+revision = '50e99c66e72f'
+down_revision = '3a33bc153e7e'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('assoc_signal_instance_tags',
```
Comment (kevgliss) [inline src/dispatch/database/revisions/tenant/versions/2024-04-22_50e99c66e72f.py:21]: It doesn't look like this table was added in code? I would double check the migration was run against an up-to-date database, as I wouldn't expect these migrations to be run given the code changes.

File: src/dispatch/incident/service.py [hunk 4]
```diff
@@ -81,6 +80,11 @@ def get_by_name(*, db_session, project_id: int, name: str) -> Optional[Incident]
     )
 
 
+def get_all_by_incident_type(*, db_session, incident_type_id: int) -> List[Optional[Incident]]:
```
Comment (kevgliss) [inline src/dispatch/incident/service.py:83]: nit: s/incident/incidents

File: tests/incident_type/test_incident_type_service.py [hunk 5]
```diff
@@ -61,6 +61,47 @@ def test_update(session, incident_type):
     assert incident_type.name == name
 
 
+def test_update_cost_model(session, incident, incident_type, cost_model, incident_cost_type):
```
Comment (kevgliss) [inline tests/incident_type/test_incident_type_service.py:64]: Yay! Tests 🎉

**Review by kevgliss — DISMISSED** [review]

File: src/dispatch/database/revisions/tenant/versions/2024-04-22_50e99c66e72f.py (hunk 3 above)
Comment (metroid-samus) [reply src/dispatch/database/revisions/tenant/versions/2024-04-22_50e99c66e72f.py]: These entries still occur with an up-to-date database. I've now manually removed them from this PR.

File: src/dispatch/database/revisions/core/versions/2024-04-22_83ec59cd3e4c.py (hunk 2 above)
Comment (metroid-samus) [reply src/dispatch/database/revisions/core/versions/2024-04-22_83ec59cd3e4c.py]: Done. Removed this file.

File: src/dispatch/incident/service.py (hunk 4 above)
Comment (metroid-samus) [reply src/dispatch/incident/service.py:83]: I named this function to be consistent with the other functions in this file (e.g. `get_all_by_status()` and `get_all_last_x_hours_by_status()`).

**Review by wssheldon — APPROVED** [review]

**Pull request merged**